### PR TITLE
Fix #806

### LINKED ISSUE #806: [SDK] max packet size is never set

### PR DESCRIPTION
**Pull request by JaylinYu**

### CODE REVIEW

**Review by RanMaoyi — APPROVED** [review]

File: src/sp/protocol/mqtt/mqtt_parser.c [hunk 1]
```diff
@@ -765,8 +765,9 @@ nmq_connack_encode(nng_msg *msg, conn_param *cparam, uint8_t reason)
 	nni_msg_append(msg, &reason, 1);
 
 	if (cparam->pro_ver == MQTT_PROTOCOL_VERSION_v5) {
-		// TODO set properties if necessary. Null by default.
-		encode_properties(msg, NULL, CMD_CONNACK);
+		// TODO change properties if necessary.
+		// modify it in nego_cp while init cparam
+		encode_properties(msg, cparam->properties, CMD_CONNACK);
```
Comment (wanghaEMQ) [inline src/sp/protocol/mqtt/mqtt_parser.c:770]: Not allo properties in connect are allowed in connack

**Review by wanghaEMQ — CHANGES_REQUESTED** [review]

File: src/sp/protocol/mqtt/mqtt_parser.c (hunk 1 above)
Comment (wanghaEMQ) [reply src/sp/protocol/mqtt/mqtt_parser.c:770]: Don't be lazy ^_^

Comment (JaylinYu) [reply src/sp/protocol/mqtt/mqtt_parser.c:770]: oh shit

File: src/mqtt/transport/tcp/mqtt_tcp.c [hunk 2]
```diff
@@ -203,8 +203,9 @@ mqtt_tcptran_pipe_init(void *arg, nni_pipe *npipe)
 	p->npipe = npipe;
 	// nni_lmq_init(&p->rslmq, 10240);
 	p->busy = false;
-	p->packmax = 0xFFFF;
-	p->qosmax  = 2;
+	// set max value by default
+	p->packmax == 0 ? p->packmax = (uint32_t)0xFFFFFFFF : p->packmax;
```
[the remark this answers is not in the document]
Comment (JaylinYu) [reply src/mqtt/transport/tcp/mqtt_tcp.c:207]: yes

**Pull request merged**